[feliciahmq] iP

feliciahmq opened the pull request:
### Claudia

> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Claudia frees your mind of having to remember things you need to do. It's, 

- text-based
- easy to learn
- ~FAST~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here](https://github.com/feliciahmq/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you  😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
---
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

``` java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

feliciahmq committed:
Add Claudia.java

feliciahmq committed:
Restructure greet and exit

feliciahmq committed:
Add support for echo

feliciahmq committed:
Add support for list

feliciahmq committed:
Add Mark as Done

feliciahmq committed:
Clean code

feliciahmq committed:
Add ToDo.java Deadline.java Event.java

feliciahmq committed:
Add TextUI Testing

feliciahmq committed:
Add all Exceptions

feliciahmq committed:
Clean code

feliciahmq committed:
Add delete command

feliciahmq committed:
Add Command enums

feliciahmq committed:
Add EmptyListException in Delete command

feliciahmq committed:
Add Claudia Guide in README.md

feliciahmq committed:
Refactor exception handling to use custom ClaudiaException subclasses

feliciahmq committed:
Clean code

feliciahmq committed:
Refactor files into folders

feliciahmq committed:
Refactor Claudia.java to implement commands as separate classes extending Command.java

feliciahmq committed:
Fix incorrect output in runtest.sh

feliciahmq committed:
Add save functionality

feliciahmq committed:
Merge branch-Level-7 into master

feliciahmq committed:
Add DateTimeParser

feliciahmq committed:
Update TextUiTesting

feliciahmq committed:
Merge branch-Level-8 into master

feliciahmq committed:
Add Parser.java Ui.java

feliciahmq committed:
Merge branch-A-MoreOOP into master

feliciahmq committed:
Add claudia package

feliciahmq committed:
Merge branch-A-Packages into master

feliciahmq committed:
Merge add-gradle-support branch into master

feliciahmq committed:
Add Gradle setup

feliciahmq committed:
Add Junit tests

feliciahmq committed:
Merge branch-A-JUnit into master

feliciahmq committed:
Add JavaDoc comments

feliciahmq committed:
Edit for coding standard

feliciahmq committed:
Edit import statements

feliciahmq committed:
Add Find command

feliciahmq committed:
Merge branch-A-CodingStandard into master

feliciahmq committed:
Merge branch-A-JavaDoc into master

feliciahmq committed:
Merge branch-Level-9 into master

feliciahmq committed:
Edit CheckStyle

feliciahmq committed:
Merge branch-A-CheckStyle into master

feliciahmq committed:
Create Claudia GUI

feliciahmq committed:
Fix Claudia GUI connection

feliciahmq committed:
Edit CSS

NightlyAffair commented on src/main/java/claudia/ui/Claudia.java:
```diff
@@ -0,0 +1,69 @@
+    private Ui ui;
+
+    /**
+     * Constructs a Friday chatbot, initializing user interface, storage and TaskList.
```
I think meant to say constructs a Claudia chatbot?

feliciahmq replied:
Agreed.

feliciahmq committed:
Add images

NightlyAffair commented on src/main/java/claudia/ui/Claudia.java:
```diff
@@ -0,0 +1,69 @@
+    /**
+     * Runs Claudia chatbot, reading and executing user command
+     * till exit command is given by user.
+     */
```
I think this is a great structure for the chatbot, keeps everything easy to track and trace 👍

feliciahmq committed:
Merge branch-Level-10 into master

NightlyAffair commented on src/main/java/claudia/ui/Ui.java:
```diff
@@ -0,0 +1,173 @@
+        System.out.println(" Got it. I've added this task:");
+        System.out.println("  " + event.toString());
+        System.out.printf(" Now you have %d tasks in the list.\n", tasks.size());
+    }
```
Why not have one method as below to handle all 3 cases by taking in a task?

```
public void showEvent(TaskList tasks, Task task) {
        System.out.println(" Got it. I've added this task:");
        System.out.println("  " + task.toString());
        System.out.printf(" Now you have %d tasks in the list.\n", tasks.size());
    }
```

feliciahmq replied:
Agreed, I will add a new method to reduce repetition of code.

feliciahmq committed:
Add var-args to find command

feliciahmq committed:
Merge branch-A-Varargs into master

NightlyAffair reviewed (commented):
I think very clean code with methods in logical places allowing for very structured packages and classes.

feliciahmq committed:
Edit README.md

feliciahmq committed:
Edit README.md

feliciahmq committed:
Update JavaDoc comments

feliciahmq committed:
Use Assertions

feliciahmq committed:
Refactor UI Class

The UI class contains several methods that format and display messages for task additions.

Unifying all those duplicated code improves the code quality.

Let's extract those duplicated code blocks into separate method called addTask(). Doing so will reduce repetition and improve consistency.

feliciahmq committed:
Use Streams in FindCommand

The FindCommand constructor and findTasksByKeyword() previously used
nested loops to process keywords, making the code more imperative.

Unifying these operations with Java Streams makes code more declarative
and concise.

Let's
* replace loops with streams
* use methods like filter() and flatMap()

These will increase readibility of code.

feliciahmq committed:
Merge pull request #1 from feliciahmq/branch-A-Assertions

Use Assertions

feliciahmq committed:
Merge master into branch-A-CodeQuality

feliciahmq committed:
Merge master into branch-A-Streams

feliciahmq committed:
Edit Checkstyle

feliciahmq committed:
Merge pull request #3 from feliciahmq/branch-A-CodeQuality

Refactor UI Class

feliciahmq committed:
Merge pull request #2 from feliciahmq/branch-A-Streams

Use Streams in FindCommand

feliciahmq committed:
Merge branch-A-Streams into master

feliciahmq committed:
Set up CI

feliciahmq committed:
Merge branch-A-CI into master

feliciahmq committed:
Fix Checkstyle Main

feliciahmq committed:
Edit gradle.yml

Golddirio commented on src/main/java/claudia/parser/Parser.java:
```diff
@@ -0,0 +1,107 @@
+         * @param command The command string.
+         * @return The valid CommandType or Unknown if invalid.
+         */
+        public static CommandType fromString(String command) {
```
Maybe the naming can be clearer?

feliciahmq replied:
Agreed!

Golddirio commented on src/main/java/claudia/task/Deadline.java:
```diff
@@ -0,0 +1,65 @@
+     *
+     * @return The file format representation of the Deadline task.
+     */
+    public String fileFormat() {
```
Maybe the naming can use a verb instead?

feliciahmq replied:
Sounds good! Thanks for feedback :)

feliciahmq committed:
Add C-Tagging

feliciahmq committed:
Merge branch-BCD-Extension into master

feliciahmq committed:
Edit Checkstyle

feliciahmq committed:
Edit Checkstyle Test

feliciahmq committed:
Fix Checkstyle error

feliciahmq committed:
Add CSS to error response

feliciahmq committed:
Add emojis

feliciahmq committed:
Merge branch-A-BetterGui into master

feliciahmq committed:
Merge branch-A-Personality into master

feliciahmq committed:
Add DateValidationException

feliciahmq committed:
Merge branch-A-MoreErrorHandling into master

feliciahmq committed:
Add EventTest

feliciahmq committed:
Merge branch-A-MoreTesting into master

feliciahmq committed:
Fix Checkstyle

feliciahmq committed:
Improve code quality

feliciahmq committed:
Merge branch-A-CodeQuality into master

feliciahmq committed:
Edit README.md

feliciahmq committed:
Merge branch-A-UserGuide into master

feliciahmq committed:
Add acknowledgements

feliciahmq committed:
Edit README.md
